# 01. Importing libraries

In [1]:
#import libraries
import pandas as pd
import numpy as np
import os

# 02. Importing data from orders.csv

In [2]:
#creating a path
path= r'C:\Users\amore\Documents\Data Analyst course\Data Immersion\Achievement 4_python\Instacard_basket_analysis_29_01_2023'

In [3]:
path

'C:\\Users\\amore\\Documents\\Data Analyst course\\Data Immersion\\Achievement 4_python\\Instacard_basket_analysis_29_01_2023'

In [4]:
#creating a list to exclude eval_set
vars_list = ['order_id', 'user_id','order_number', 'order_dow', 'order_hour_of_day','days_since_prior_order']

In [5]:
#importing the data from orders.csv
df=pd.read_csv(os.path.join(path, '02 Data','Original Data','orders.csv'), usecols=vars_list, index_col=False)

In [6]:
df.head()

order_id  user_id  order_number  order_dow  order_hour_of_day  \
0   2539329        1             1          2                  8   
1   2398795        1             2          3                  7   
2    473747        1             3          3                 12   
3   2254736        1             4          4                  7   
4    431534        1             5          4                 15   

   days_since_prior_order  
0                     NaN  
1                    15.0  
2                    21.0  
3                    29.0  
4                    28.0

# 03 Importing dat from products.csv

In [7]:
#Importing the data from products.csv
df_prods=pd.read_csv(os.path.join(path, '02 Data','Original Data','products.csv'), index_col=False)

In [8]:
#Printing the first 20 rows and the last 35 rows of df_prods
df_prods.head(20), df_prods.tail(35)

(    product_id                                       product_name  aisle_id  \
 0            1                         Chocolate Sandwich Cookies        61   
 1            2                                   All-Seasons Salt       104   
 2            3               Robust Golden Unsweetened Oolong Tea        94   
 3            4  Smart Ones Classic Favorites Mini Rigatoni Wit...        38   
 4            5                          Green Chile Anytime Sauce         5   
 5            6                                       Dry Nose Oil        11   
 6            7                     Pure Coconut Water With Orange        98   
 7            8                  Cut Russet Potatoes Steam N' Mash       116   
 8            9                  Light Strawberry Blueberry Yogurt       120   
 9           10     Sparkling Orange Juice & Prickly Pear Beverage       115   
 10          11                                  Peach Mango Juice        31   
 11          12                         

In [9]:
#Print the names of the columns of df_prods
df_prods.columns

Index(['product_id', 'product_name', 'aisle_id', 'department_id', 'prices'], dtype='object')

In [10]:
#Print the number of rows and columns of df_prods
df_prods.shape

(49693, 5)

# 04. ANSWERS

In [11]:
#obtaining descriptive statistics of df_prods
df_prods.describe()

product_id      aisle_id  department_id        prices
count  49693.000000  49693.000000   49693.000000  49693.000000
mean   24844.345139     67.770249      11.728433      9.994136
std    14343.717401     38.316774       5.850282    453.519686
min        1.000000      1.000000       1.000000      1.000000
25%    12423.000000     35.000000       7.000000      4.100000
50%    24845.000000     69.000000      13.000000      7.100000
75%    37265.000000    100.000000      17.000000     11.200000
max    49688.000000    134.000000      21.000000  99999.000000

In [12]:
#obtaining the data types of the columns of df_prods
df_prods.dtypes

product_id         int64
product_name      object
aisle_id           int64
department_id      int64
prices           float64
dtype: object

What is the maximum value of the aisle_id column?

It's 134

What is the data type of the "department_id" column? 

It is int64

# Pruebas 4.4

In [13]:
df.rename(columns={'order_dow':'orders_day_of_week'}, inplace=True)

In [14]:
df.head()

order_id  user_id  order_number  orders_day_of_week  order_hour_of_day  \
0   2539329        1             1                   2                  8   
1   2398795        1             2                   3                  7   
2    473747        1             3                   3                 12   
3   2254736        1             4                   4                  7   
4    431534        1             5                   4                 15   

   days_since_prior_order  
0                     NaN  
1                    15.0  
2                    21.0  
3                    29.0  
4                    28.0

In [15]:
df['order_id'] = df['order_id'].astype('str')

In [16]:
df.describe()

user_id  order_number  orders_day_of_week  order_hour_of_day  \
count  3.421083e+06  3.421083e+06        3.421083e+06       3.421083e+06   
mean   1.029782e+05  1.715486e+01        2.776219e+00       1.345202e+01   
std    5.953372e+04  1.773316e+01        2.046829e+00       4.226088e+00   
min    1.000000e+00  1.000000e+00        0.000000e+00       0.000000e+00   
25%    5.139400e+04  5.000000e+00        1.000000e+00       1.000000e+01   
50%    1.026890e+05  1.100000e+01        3.000000e+00       1.300000e+01   
75%    1.543850e+05  2.300000e+01        5.000000e+00       1.600000e+01   
max    2.062090e+05  1.000000e+02        6.000000e+00       2.300000e+01   

       days_since_prior_order  
count            3.214874e+06  
mean             1.111484e+01  
std              9.206737e+00  
min              0.000000e+00  
25%              4.000000e+00  
50%              7.000000e+00  
75%              1.500000e+01  
max              3.000000e+01

In [17]:
df['order_id'].dtype

dtype('O')

In [18]:
df['order_id'].dtype

dtype('O')

# Part 4.4

In [20]:
#importing departments.csv data set
df_dep = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'departments.csv'), index_col=False)

In [23]:
df_dep.head()

department_id       1      2       3        4        5              6  \
0    department  frozen  other  bakery  produce  alcohol  international   

           7     8                9  ...            12      13         14  \
0  beverages  pets  dry goods pasta  ...  meat seafood  pantry  breakfast   

             15          16         17      18      19    20       21  
0  canned goods  dairy eggs  household  babies  snacks  deli  missing  

[1 rows x 22 columns]

In [24]:
df_dep.T

0
department_id       department
1                       frozen
2                        other
3                       bakery
4                      produce
5                      alcohol
6                international
7                    beverages
8                         pets
9              dry goods pasta
10                        bulk
11               personal care
12                meat seafood
13                      pantry
14                   breakfast
15                canned goods
16                  dairy eggs
17                   household
18                      babies
19                      snacks
20                        deli
21                     missing

In [25]:
df_dep_t=df_dep.T

In [26]:
df_dep_t

0
department_id       department
1                       frozen
2                        other
3                       bakery
4                      produce
5                      alcohol
6                international
7                    beverages
8                         pets
9              dry goods pasta
10                        bulk
11               personal care
12                meat seafood
13                      pantry
14                   breakfast
15                canned goods
16                  dairy eggs
17                   household
18                      babies
19                      snacks
20                        deli
21                     missing

In [27]:
df_dep_t.reset_index()

index                0
0   department_id       department
1               1           frozen
2               2            other
3               3           bakery
4               4          produce
5               5          alcohol
6               6    international
7               7        beverages
8               8             pets
9               9  dry goods pasta
10             10             bulk
11             11    personal care
12             12     meat seafood
13             13           pantry
14             14        breakfast
15             15     canned goods
16             16       dairy eggs
17             17        household
18             18           babies
19             19           snacks
20             20             deli
21             21          missing

In [30]:
new_header = df_dep_t.iloc[0]

In [31]:
new_header

0    department
Name: department_id, dtype: object

In [32]:
df_dep_t_new = df_dep_t[1:]

In [34]:
df_dep_t_new.columns=new_header

In [35]:
df_dep_t_new

department_id       department
1                       frozen
2                        other
3                       bakery
4                      produce
5                      alcohol
6                international
7                    beverages
8                         pets
9              dry goods pasta
10                        bulk
11               personal care
12                meat seafood
13                      pantry
14                   breakfast
15                canned goods
16                  dairy eggs
17                   household
18                      babies
19                      snacks
20                        deli
21                     missing